x-pack/plugins/security_solution/public/common/utils/route/mocks.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RouteSpyState } from './types';
+
+type Action = 'PUSH' | 'POP' | 'REPLACE';
+
+const pop: Action = 'POP';
+
+const generateDefaultLocationMock = () => ({
```

Comment:
Thanks for making all of these functional btw.

Reply:
NP. Yeah, its still unclear to me what was causing the weird failures and why they started happening now. What made it even harder to debug was that if I ran the single test that failed locally, it would run successful. it was only when I ran the entire suite (security solution) that I was getting failures, so I started to remove the shared/mutable mocks between tests suites and that seemed to have done it.